src/libraries/System.Text.RegularExpressions/gen/UpgradeToGeneratedRegexCodeFixer.cs

```diff
@@ -298,7 +298,8 @@ static RegexOptions GetRegexOptionsFromArgument(ImmutableArray<IArgumentOperatio
                 }
                 else if (argument.Value.ConstantValue.Value is string str && str.Contains('\\'))
                 {
-                    return SyntaxFactory.ParseExpression($"@\"{str}\"");
+                    string escapedVerbatimText = str.Replace("\"", "\"\"");
+                    return SyntaxFactory.ParseExpression($"@\"{escapedVerbatimText}\"");
```

Comment:
Is there no equivalent in Roslyn to generator.LiteralExpression that would tell it to emit the string as a verbatim string rather than us manually doing it?

Reply:
Not that I could find. We have to build the verbatim syntax with `ParseExpression` or `SyntaxFactory` like this:
```cs
SyntaxFactory.LiteralExpression(
    SyntaxKind.StringLiteralExpression,
    SyntaxFactory.Literal(str, $"@\"{escapedVerbatimText}\""));
```

This still requires us to manually escape the string and build the verbatim string.

Reply:
@333fred, is there a better way?

Reply:
We have an API for non-verbatims (`SymbolDisplay.FormatLiteral`), but not verbatims, AFAIK.

Reply:
Ok, thanks